kbase-extension/static/kbase/js/common/jobs.js

```
@@ -1,37 +1,129 @@
+     * 4. It has a 'status' key, which appears in the array validJobStates
+     *
+     * There are other required fields, but these checks are sufficient for the UI.
+     *
+     * See execution_engine2/lib/execution_engine2/db/models/models.py in the
+     * execution_engine2 repo for the full job spec.
+     *
+     * This function should be updated to stay in sync with ee2's output.
+     *
+     * @param {object} jobState
+     * @returns {boolean} true|false
+     */
+    function isValidJobStateObject(jobState) {
+        const requiredProperties = ['job_id', 'created', 'status'];
+        return (
+            jobState !== null &&
+            typeof jobState === 'object' &&
+            requiredProperties.every((prop) => prop in jobState) &&
+            validJobStates.includes(jobState.status)
+        );
+    }
+
+    /**
+     * Given a job status string, ensure that it is valid
+     *
+     * @param {*} status
+     * @returns {boolean} true|false
+     */
+    function isValidJobStatus(status) {
+        return status && validJobStates.includes(status);
+    }
+
+    /**
+     * A jobInfo object should have the minimal structure
+     *
+     * {
+     *      job_id: ...
+     *      job_params: [
+     *          {
+     *              param_one: 'value one',
+     *              param_two: 'value two',
+     *              ...
+     *          }
+     *      ]
+     * }
+     *
+     * This function should be updated to stay in sync with ee2's output.
+     *
+     * @param {object} jobInfo
+     * @returns {boolean} true|false
+     */
+    function isValidJobInfoObject(jobInfo) {
+        try {
+            if (
+                'job_id' in jobInfo &&
+                Object.prototype.toString.call(jobInfo.job_params[0]) === '[object Object]' &&
+                Object.keys(jobInfo.job_params[0]).length > 0
+            ) {
+                return true;
+            }
+        // eslint-disable-next-line no-empty
+        } catch (err) {}
```

we're just checking whether the job info object is valid; we don't really care how it deviates from the expected structure, so this error can be ignored